File: teams/Angela_Matevosyan/Python/31_01_2022/fibonacci.py
n = int(input("Enter the number: "))
def fibonacciRec(n):
    if n <= 0:
        print("Incorrect input")
    elif n == 1:
        return 0
    elif n == 2:
        return 1
    else:
        return fibonacciRec(n-1) + fibonacciRec(n-2)
print("Is called function with recursion.")
print("The nth Fibonacci number is", fibonacciRec(n))

def fibonacciLoop(n):
    if n <= 0:
        return "Incorrect input"
    data = [0, 1]
    if n > 2:
        for i in range(2, n):
            data.append(data[i-1] + data[i-2])
    return data[n-1]
print("Is called function with loop.")
print("The nth Fibonacci number is", fibonacciLoop(n))

File: teams/Margarita_Hambaryan/C++/15_12_2021/fibonacci.cpp
#include <iostream>
int fibonacci(int n) {
	if (n <= 1) {
		return n;
	}
	return fibonacci(n - 1) + fibonacci(n - 2);
}

int main() {
	int n;
	std::cout << "Please input the number of fibonacci sequence: ";
	std::cin >> n;
	if (n >=1) {
		std::cout << "The "<< n <<"th fibonacci number is: " << fibonacci(n - 1) << std::endl;
	} else if (n == 0) {
		std::cout << "The 0th fibonacci number is 0\n";
	}
	return 0;
}
